src/browser/BLECentralPlugin.js

```diff
@@ -1,63 +1,194 @@
+                        server: server
+                    })
+                    connectSuccess(device);
+                });
+            }).catch(connectFailure);
+        }
+    },
+    disconnect: function(device_id, disconnectSuccess, disconnectFailure) {
+        if (this.devices.has(device_id)) {
+            var device = this.devices.get(device_id).device;
+            this.devices.delete(device_id);
+            if (device.gatt.connected) {
+                device.gatt.disconnect();
+                disconnectSuccess(device);
+            } else {
+                disconnectFailure("Device not connected");
+            }
+        } else if (disconnectFailure) {
+            disconnectFailure();
+        }
     },
     read: function(device_id, service_uuid, characteristic_uuid, success, failure) {
-        notSupported();
-        if (failure) failure();
+       if (this.devices.has(device_id)) {
+            this.devices.get(device_id).server.getPrimaryService(service_uuid).then(service => {
+                return service.getCharacteristic(characteristic_uuid);
+            }).then(characteristic => {
+                return characteristic.readValue();
+            }).then(result => {
+                success(result);
+            }).catch(error => {
+                if (failure) failure(error);
+            });
+        } else if (failure) { 
+          failure();
+        }
     },
     readRSSI: function(device_id, success, failure) {
         notSupported();
         if (failure) failure();
     },
     write: function(device_id, service_uuid, characteristic_uuid, data, success, failure) {
-        notSupported();
-        if (failure) failure();
+        if (this.devices.has(device_id)) {
+            this.devices.get(device_id).server.getPrimaryService(service_uuid).then(service => {
+                return service.getCharacteristic(characteristic_uuid);
+            }).then(characteristic => {
+                return characteristic.writeValue(data);
+            }).then(result => {
+                success(result);
+            }).catch(error => {
+                if (failure) failure(error);
+            });
+        } else if (failure) { 
+          failure();
+        }
     },
     writeWithoutResponse: function(device_id, service_uuid, characteristic_uuid, data, success, failure) {
-        notSupported();
-        if (failure) failure();
+        if (this.devices.has(device_id)) {
+            this.devices.get(device_id).server.getPrimaryService(service_uuid).then(service => {
+                return service.getCharacteristic(characteristic_uuid);
+            }).then(characteristic => {
+                return characteristic.writeValue(data);
+            }).then(result => {
+                success(result);
+            }).catch(error => {
+                // Ignore Error 
+                // failure();
+            });
+        } else if (failure) { 
+            // Ignore Error 
```

Comment:
I think it should still be considered a failure if the device/service/characteristic don't exist...